app/services/settings/authentications/upsert.rb

```diff
@@ -0,0 +1,72 @@
+module Settings
+  module Authentications
```

Comment:
the plural here is a bit weird. "Settings" feels like a collective noun, "authentications" doesn't. What about `Settings::Authentication`, but I guess we can't because that's the name of the model :-( Maybe we can switch it up to `Authentication::Settings` ? After all everything about authentication is in `services/authentication`, except... the settings?

Reply:
Hmm, I brought up this plural thing before. This is not only about the name-clash, but also about following Rails conventions as to what gets pluralized (i.e. most of our services use pluralized models). That said, I like `Authentication::Settings::Upsert`, I think I'll go with that.

Reply:
This is now in `app/services/authentication/settings_upsert.rb` and the service is `Authentication::SettingsUpsert`.